src/jetstream.rs

```diff
@@ -559,15 +646,15 @@ impl Consumer {
     /// `ConsumerInfo` that may have been returned
     /// from the `nats::Connection::list_consumers`
     /// iterator.
-    pub fn from_consumer_info(ci: ConsumerInfo, nc: NatsClient) -> io::Result<Consumer> {
-        Consumer::existing::<String, ConsumerConfig>(nc, ci.stream_name, ci.config)
+    pub fn from_consumer_info(ci: ConsumerInfo, js: JetStream) -> io::Result<Consumer> {
+        Consumer::existing::<String, ConsumerConfig>(js, ci.stream_name, ci.config)
     }
 
     /// Instantiate a `JetStream` `Consumer`. Performs a check to see if the consumer
     /// already exists, and creates it if not. If you want to use an existing
     /// `Consumer` without this check and creation, use the `Consumer::existing`
     /// method.
-    pub fn create_or_open<S, C>(nc: NatsClient, stream: S, cfg: C) -> io::Result<Consumer>
+    pub fn create_or_open<S, C>(js: JetStream, stream: S, cfg: C) -> io::Result<Consumer>
```

Comment:
This looks like something we might like to clean up, as it's 
creating consumer is in JetStream struct,
reusing existing consumer is in Consumer struct.

I guess that it means "Using existing consumers is part of Consumer struct, Creating consumers is part of JetStream (managemnet)".

But then we have fn  that reuses or creates Consumer. Nice idempotency feature, but becaues of that split, neither Consumer nor JS look like a good place for it (it's mixed jetstream management with jetstream usage).

We might preserve those functions here, but I would exponse then also in JetStream struct.